packages/smooth_app/lib/background/background_task_manager.dart

```diff
@@ -100,25 +119,51 @@ class BackgroundTaskManager {
         block ? '' : null,
       );
 
-  /// Runs all the pending tasks, until it crashes.
+  /// Runs all the pending tasks, and then smoothly ends.
   Future<void> run() async {
-    if (!_canStartNow()) {
+    await _run();
+    await _justFinished();
+  }
+
+  /// Runs all the pending tasks.
+  ///
+  /// If a task fails, we continue with the other tasks: and we'll retry the
+  /// failed tasks later.
+  /// If a task fails and another task with the same stamp comes after,
+  /// we can remove the failed task from the list: it would have been
+  /// overwritten anyway.
+  Future<void> _run() async {
+    if (!await _canStartNow()) {
       return;
     }
-    AbstractBackgroundTask? nextTask;
-    try {
-      while ((nextTask = await _getNextTask()) != null) {
-        if (blocked) {
+    final List<AbstractBackgroundTask> tasks = await _getAllTasks();
+    final Map<String, String> failedTaskFromStamps = <String, String>{};
+    for (final AbstractBackgroundTask task in tasks) {
+      if (blocked) {
+        return;
+      }
+      final String stamp = task.stamp;
+      final String? previousFailedTaskId = failedTaskFromStamps[stamp];
+      if (previousFailedTaskId != null) {
+        // there was a similar task that failed previously and we can dismiss it
+        // as the current one would overwrite it.
+        // not only will we spare a to-be-overwritten call, but we avoid the
+        // "save latest change" and then "save initial change" dilemma.
+        _debugPrint('removing failed task $previousFailedTaskId');
+        await _remove(previousFailedTaskId);
+        failedTaskFromStamps.remove(stamp);
+      }
+      try {
+        await _runTask(task);
+      } catch (e) {
```

Comment:
It looks like `TimeoutException` is not appropriate here: `Thrown when a scheduled timeout happens while waiting for an async result.`.

`SocketException` could be a good idea, but honestly that's not the added value of this PR. And I don't want all the next tasks to be ignored just because one task had a `SocketException` that was not about the internet/the server being down.

To be studied and perhaps PR'ed later, but what is urgent now is the "stamping".